## Практическое задание 1

#### Запишем сразу все функции для комбинаторики

Число **сочетаний** из $n$ элементов по $k$ элементов в каждом (в сочетаниях порядок не важен):

$$C_n^k = \frac{n!}{k!(n - k)!}$$

In [1]:
from math import factorial

In [2]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

Число **размещений** из $n$ элементов по $k$ элементов в каждом. При размещениях порядок важен, поэтому вариантов размещения может быть больше, чем сочетаний при заданных $k$  и $n$.

$$A_n^k = \frac{n!}{(n - k)!}$$

In [3]:
def arrangements(n, k):
    return int(factorial(n) / factorial(n - k))

Число **перестановок** из $n$ элементов — при перестановках важен порядок, но отличие от размещений в том, что применяются все имеющиеся $n$ элементов:

$$P_n = n!$$

In [4]:
def permutations(n):
    return int(factorial(n))

### _1_ Из колоды в 52 карты извлекаются случайным образом 4 карты.  
   ##### a) Найти вероятность того, что все карты – крести.  
Ответ:  
Рассчитаем сколькими способами можно вытащить 4 карты из колоды, состоящей из 52 карт

In [5]:
combinations(52, 4)

270725

Рассчитаем сколькими способами можно вытащить 4 крести из 13 возможных 

In [6]:
combinations(13, 4)

715

По формуле $$P(A) = \frac{m}{n}$$ найдем ввероятность

In [7]:
P = combinations(13, 4)/combinations(52, 4)
P

0.0026410564225690276

##### Ответ:  Вероятность того, что все карты – крести = 0.0026410564225690276

##### б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

В данной сиуации возможно 4 варианта: 1, 2, 3 или 4 туза. Рассмтрим каждый случай и сложим вероятности.

Рассмотрим первую ситуацию, когда выбирается один туз из четырех.

Число таких сочетаний будет равно:

In [8]:
combinations(4, 1)

4

Остальные три карты выбираются из 48 карт  — тузы не рассматривается, так как он уже выбраны:

In [9]:
combinations(48, 3)

17296

И в этом случае будет такое число сочетаний, когда из колоды, в которой 52 карты, выбраны четыре, одна из которых — туз:

In [10]:
4 * 17296

69184

Рассмотрим вторую ситуацию, когда выбирается два туза из четырех.

Число таких сочетаний будет равно:

In [11]:
combinations(4, 2)

6

Остальные две карты выбираются из 48 карт  — тузы не рассматривается, так как они уже выбраны:

In [12]:
combinations(48, 2)

1128

И в этом случае будет такое число сочетаний, когда из колоды, в которой 52 карты, выбраны четыре, две из которых — туз:

In [13]:
6 * 1128

6768

Рассмотрим третью ситуацию, когда выбираются три туза из четырех.

Число таких сочетаний будет равно:

In [14]:
combinations(4, 3)

4

Оставшаяся карта выбираются из 48 карт  — тузы не рассматривается, так как они уже выбраны:

In [15]:
combinations(48, 1)

48

И в этом случае будет такое число сочетаний, когда из колоды, в которой 52 карты, выбраны четыре, три из которых — туз:

In [16]:
4 * 48

192

Рассмотрим последнюю ситуацию, когда выбираются четыре туза из четырех.

Число таких сочетаний будет равно:

In [17]:
combinations(4, 4)

1

Оставшаяся карта не выбираются, так как все карты уже выбраны

И в этом случае будет такое число сочетаний, когда из колоды, в которой 52 карты, выбраны четыре, и все тузы: 1

Оющее количесво исходов равно:

In [18]:
combinations(48, 4)

194580

Теперь найдем и сложим вероятности:

In [19]:
69184/270725+6768/270725+192/270725+1/270725

0.2812632745405855

##### Ответ:  Вероятность того, что среди 4-х карт окажется хотя бы один туз = 0.2812632745405855

### _2_ На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

Применяем формулу: Р = число исходов, благоприятствующих нашему событию / общее число исходов.  
Код один, поэтому число исходов, благоприятствующих нашему событию = 1.  
В данной задаче порядок не имеет значения, поскольку кнопки нажиаются одновременно. Следовательно, общее число исходов равычисляем по формуле сочетаний. 

In [20]:
combinations(10, 3)

120

Далее получаем P = 1/120

### _3_ В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

Применяем формулу: Р = число исходов, благоприятствующих нашему событию / общее число исходов.  
Число благоприятных исходов равно числу способов, которыми можно извлечь 3 детали из 9 окрашенных. Порядок значения не имеет, поэтому используем формулу сочетаний.
Общее число исходов равно числу способов, которыми можно извлечь 3 детали из 15. Порядок значения не имеет, поэтому используем формулу сочетаний. Далее считаем по формуле:

In [21]:
P = combinations(9, 3)/combinations(15, 3)
P

0.18461538461538463

##### Ответ:  Вероятность того, что все извлеченные детали окрашены = 0.18461538461538463

### _4_ В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

Применяем формулу: Р = число исходов, благоприятствующих нашему событию / общее число исходов.  
Число благоприятных исходов равно числу способов, которыми можно купить 2 билета из 2. Порядок значения не имеет, поэтому используем формулу сочетаний.
Общее число исходов равно равно числу способов, которыми можно купить 2 билета из 100. Порядок значения не имеет, поэтому используем формулу сочетаний. Далее считаем по формуле:

In [22]:
P = combinations(2, 2)/combinations(100, 2)
P

0.00020202020202020202

##### Ответ:  Вероятность того, что 2 приобретенных билета окажутся выигрышными = 0.00020202020202020202